### DISCUSSION TITLE
Something similar or equivalent for iOS

### DISCUSSION THREAD

**Discussion by guaifi** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#455] [Q&A]
I would like to know if there is something equivalent but that works for ipad. Ipad Pro M1 exactly.

I like the app. Great job!

Thank you very much

**Comment by blueboy31** [comment]
I have an iPad Pro and am also interested in this. Please look into AlDente for iPadOS and iOS!

**Reply by lich4** [reply]
https://github.com/lich4/ChargeLimiter

**Comment by ElhemEnohpi** [comment]
I'm not sure, but I would think that it's not possible for an iOS app to control the battery charging. One alternative is to use one of the many alarm apps. I use one called "battery charge alarm". It has ads, but otherwise works ok, there might be better ones. They don't limit charging automatically, but give an alarm sound when it reaches your set level. Another option is a hardware device called "chargie" that has an app, and does shut off charging automatically. But it's a bit expensive. Though less expensive than a new battery I guess.

[edit, about a year later: I'm really sure there's no way to do this purely in software on iOS without a jailbreak, and I got tired of having the battery alarm app going off all the time.  I ended up buying a Chargie, which works great. Another idea is to set up two Shortcuts, with "battery level" triggers, and use Home Assistant to turn a bluetooth smart plug on and off for the charger.]

**Reply by jqrs** [reply]
Does iOS limit to 80% overnight charge by default? My battery capacity still fell to 82% but that may have due to undercharging.. not sure

**Reply by ElhemEnohpi** [reply]
It does limit to 80% overnight, and I think it's the default. But you have to be consistent with your sleep and wake times. I could never get it to work for me. My battery is still at 96% after almost two years, keeping it between 30% and 60% charged most of the time.

**Comment by byegates** [comment]
I want the same too, I'd definitely pay for it.

**Reply by lich4** [reply]
https://github.com/lich4/ChargeLimiter

**Comment by lich4** [comment]


**Reply by ElhemEnohpi** [reply]
> iOS<=14 requires jailbreak + deb installation, starts automatically every time you jailbreak.
iOS>=15 need TrollStore+tipa installation.

I'm not really familiar with TrollStore / tipa, but from the [web page](https://iexmo.com/trollstore-helper/#TrollStore):

> You may install any app permanently assigned via the TrollStore for iOS. Furthermore, thanks to the CoreTrust flaw found by Linus Henze, it is capable of installing apps associated with different unauthorized entitlements or those that package a root helper binary.

It doesn't look like something I'd be comfortable installing. I used to use jailbreaks, but decided it wasn't worth the security risks. YMMV.

**Reply by lich4** [reply]
If you only want AppStore app, then the answer is no such thing, never, but now battery manangement is better after iPhone X and iOS16+, this kind of software for iOS is meaningless.
It requires more entitlement to implement AlDente on iOS, so Jailbreak/TrollStore is required. TrollStroll is safer than Jailbreak, It only elevate entitlement of specified apps, unless you install damaged apps

**Comment by byegates** [comment]
latest ios 17 now support limit charge limit to 80%.
ios 18 supports limits between 80%~100%

**Reply by ElhemEnohpi** [reply]
Manual charge limit in iOS is only on iPhone 15 or later.

**Reply by mthfckrim** [reply]
Charging Limiter (APP-Store) 

https://apps.apple.com/at/app/chargie-phone-charge-limiter/id1559014917


Charging Limiter (Trollstore iOS 17)

https://www.reddit.com/r/jailbreak/comments/18tjhvw/trollstore_ios_170_iphone_14_users_could_you_try/
